packages/grpc-native-core/src/server.js

```diff
@@ -664,22 +677,26 @@ function handleServerStreaming(call, handler, metadata) {
  * @param {grpc.Metadata} metadata Metadata from the client
  */
 function handleClientStreaming(call, handler, metadata) {
+  var asyncResource = createAsyncResourceWrapper('GrpcServerRequest');
   var stream = new ServerReadableStream(call, metadata, handler.deserialize);
   stream.on('error', function(error) {
     handleError(call, error);
   });
   stream.waitForCancel();
-  handler.func(stream, function(err, value, trailer, flags) {
-    stream.terminate();
-    if (err) {
-      if (trailer) {
-        err.metadata = trailer;
+  asyncResource.wrap(function() {
+    handler.func(stream, function(err, value, trailer, flags) {
```

Comment:
This use isn't making sense to me. `handler.func` seems to be getting executed in the same continuation as `handleClientStreaming` – it is called synchronously.

It doesn't seem logical to create an `AsyncResource` to wrap a synchronously executed function.

Reply:
This is to mirror what the other handler types do. I don't think there's harm in having nesting here (from the perspective of an agent listening for events with only one type of async resource, there is no concept of nesting) -- and I can't think of a good alternative without changing the behavior of the code to move the invocation `handler.func` to the next turn of the event loop. At the minimum we will definitely want a per-request `grpc.ServerRequest`-type async resource. The `grpc.Server` async resource doesn't seem strictly required at the moment but it follows the precedent set by `http`.